sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -108,49 +114,55 @@ internal static IEnumerable<BatchedLogs<T>> Batch<T>(IEnumerable<T> logEntries,
                 var memory = entry.ToMemory();
                 if (memory.Length > SingleUploadThreshold) // if single log is > 1 Mb send to be gzipped by itself
                 {
-                    MemoryStream tempStream = new MemoryStream(); // create tempStream for individual log
-                    WriteMemory(tempStream, BinaryData.FromString("[").ToMemory());
-                    WriteMemory(tempStream, memory);
-                    WriteMemory(tempStream, BinaryData.FromString("]").ToMemory());
-                    tempStream.Position = 0;
-                    yield return new BatchedLogs<T>(new List<T>{log}, BinaryData.FromStream(tempStream));
+                    // Create tempWriter and tempStream for individual log
+                    MemoryStream tempStream = new MemoryStream();
+                    Utf8JsonWriter tempWriter = new Utf8JsonWriter(tempStream);
+                    tempWriter.WriteStartArray();
+                    WriteMemory(tempWriter, memory);
+                    tempWriter.WriteEndArray();
+                    yield return new BatchedLogs<T>(new List<T>{log}, BinaryData.FromObjectAsJson(writer));
                 }
-                else if ((stream.Length + memory.Length + 1) >= SingleUploadThreshold) // if adding this entry makes stream > 1 Mb send current stream now
+                // if adding this entry makes stream > 1 Mb send current stream now
+                else if ((stream.Length + memory.Length + 1) >= SingleUploadThreshold)
                 {
-                    WriteMemory(stream, BinaryData.FromString("]").ToMemory());
-                    stream.Position = 0; // set Position to 0 to return everything from beginning of stream
-                    yield return new BatchedLogs<T>(currentLogList, BinaryData.FromStream(stream));
+                    writer.WriteEndArray();
+                    //stream.Position = 0; // set Position to 0 to return everything from beginning of stream
+                    yield return new BatchedLogs<T>(currentLogList, BinaryData.FromObjectAsJson(writer));
```

Comment:
We would still use FromStream here with the underlying memoryStream after calling Flush on the JsonWriter.